src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,7 +241,14 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            if (Sse2.IsSupported)
```

Comment:
I mentioned a similar comment in Eirik's PR - In the original code, `Sse2` was the default.
I think the `if` should be the one checking for `AdvSimd.IsSupported`, and the original `Sse2` code should be the fallback in the `else`.
This suggestion applies here and elsewhere.

Reply:
Hmmm, but it wouldn't matter right? Because the compiler will optimize away the non-relevant blocks?

Reply:
I'd rather keep it like this, `AdvSim.IsSupported` is a shimmed property in `< net5.0` targets so the compiler would emit warnings if you checked that.